embedchain/embedchain.py

```diff
@@ -188,11 +190,19 @@ def generate_prompt(self, input_query, context):
         :param context: Similar documents to the query used as context.
         :return: The prompt
         """
-        prompt = f"""Use the following pieces of context to answer the query at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.
-        {context}
-        Query: {input_query}
-        Helpful Answer:
-        """
+        if type(context) is list and len(context) > 0:
+            prompt = f"""Use the following pieces of context to answer the query at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.
+            {" | ".join(context)}
+            Query: {input_query}
+            Helpful Answer:
+            """
+        else:
+            prompt = f"""Use the following context to answer the query at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.
+            {context}
```

Comment:
```suggestion
            {context[0]}
```

Reply:
thanks for catching this.

Reply:
I was actually doing this for backwards compatibility, if someone insists on passing a single context string instead of a list. Hence the type check on 193. I guess we don't have to do that, since users should only use .add and .query and those are safe.

Anyways, we might as well remove the type check on 193.